Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Staging/Processes/Up.mo

```diff
@@ -0,0 +1,651 @@
+    annotation (Placement(transformation(extent={{0,-10},{20,10}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.Processes.Subsequences.EnableBoiler enaBoi(
+    final nBoi=nBoi,
+    final proOnTim=boiChaProOnTim)
+    "Change boiler status as per stage change required"
+    annotation (Placement(transformation(extent={{60,-10},{80,10}})));
+
+  Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Staging.Processes.Subsequences.HWIsoVal disHotWatIsoVal1(
+    final nBoi=nBoi,
+    final chaHotWatIsoTim=chaIsoValTim,
+    final iniValPos=1,
+    final endValPos=0) if isHeadered
+    "Close hot water valve for boiler being disabled"
+    annotation (Placement(transformation(extent={{150,-10},{170,10}})));
+
+  Buildings.Controls.OBC.CDL.Logical.And and3
+    "Check for completion of valve closing process and pump change process"
+    annotation (Placement(transformation(extent={{190,-10},{210,10}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Latch lat
+    "Latch to retain stage-up edge signal till the stage change process is completed"
+    annotation (Placement(transformation(extent={{-222,-10},{-202,10}})));
+
+  Buildings.Controls.OBC.CDL.Logical.TrueDelay truDel(
+    final delayTime=delBoiEna)
+    "Time delay after boiler status has been changed"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Pre pre1
+    "Logical pre block"
+    annotation (Placement(transformation(extent={{200,-60},{220,-40}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Latch lat3
+    "Hold process completion signal after pump enable process"
+    annotation (Placement(transformation(extent={{-140,-200},{-120,-180}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Latch lat4
+    "Hold process completion signal after pump disable process"
+    annotation (Placement(transformation(extent={{160,-200},{180,-180}})));
+
+  Buildings.Controls.OBC.CDL.Logical.And and4
+    "Check for pump disable completion after start of pump disable process"
+    annotation (Placement(transformation(extent={{132,-200},{152,-180}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Edge edg
+    "Detect change in process completion status and send out pulse signal"
+    annotation (Placement(transformation(extent={{230,40},{250,60}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Edge edg1
+    "Generate pulse to signal start of pump change process"
+    annotation (Placement(transformation(extent={{-100,-260},{-80,-240}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Edge edg2
+    "Generate pulse to signal start of pump change process"
+    annotation (Placement(transformation(extent={{140,-246},{160,-226}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Or or2
+    "Check for pump change proces start signal"
+    annotation (Placement(transformation(extent={{210,-250},{230,-230}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Latch lat5 if not isHeadered
```

Comment:
again the `is_...` Let's hear from Michael if this should be `is_hea` or `is_headered`.